checker/src/org/checkerframework/checker/initialization/InitializationVisitor.java

```diff
@@ -181,6 +184,19 @@ protected boolean checkContract(
                             return true;
                         }
                     }
+                } else if (fa.getReceiver() instanceof LocalVariable) {
+                    // TODO: what is the best way to handle other types of
+                    // receivers?
+                    Element elem = ((LocalVariable) fa.getReceiver()).getElement();
+                    AnnotatedTypeMirror recvType = atypeFactory.getAnnotatedType(elem);
```

Comment:
I would recommending using the store first because it contains information about more than just LocalVariables.  Then if it does not have a value, then fall back to using the AnnotatedTypeFactory if the Receiver is a LocalVariable or a FieldAccess -- the only two kinds of Receivers with an element. 

Also, I don't think getAnnotatedType(Element) will return a refined type, will it?